src/gui/tray/Window.qml

```diff
@@ -403,67 +403,59 @@ Window {
                     }
                 }
 
-                // Filler between account dropdown and header app buttons
-                Item {
-                    id: trayWindowHeaderSpacer
-                    Layout.fillWidth: true
-                }
-
-                HeaderButton {
-                    id: openLocalFolderButton
+                RowLayout {
+                    id: openLocalFolderRowLayout
+                    height: Style.trayWindowHeaderHeight
+                    width: height
+                    anchors.left: accountControlRowLayout.right
+                    anchors.top: accountControlRowLayout.top
+                    spacing: 0
+                    Layout.preferredWidth:  Style.trayWindowHeaderHeight
+                    Layout.preferredHeight: Style.trayWindowHeaderHeight
+                    
+                    HeaderButton {
+                        id: openLocalFolderButton
+                        visible: UserModel.currentUser.hasLocalFolder
+                        icon.source: "qrc:///client/theme/white/folder.svg"
+                        onClicked: UserModel.openCurrentAccountLocalFolder()
+                    }
+                                             
+                    Rectangle {
+                        id: folderStateIndicatorBackground
+                        width: Style.folderStateIndicatorSize
+                        height: width
+                        anchors.top: openLocalFolderButton.verticalCenter

```

Comment:
I believe that anchors and being inside a layout is undefined behavior and should be avoided.
You could have only the HeaderButton in the layout and the Rectangle as a child of it set at the desired position and size.

I am getting lots of warning about that in the log

> QML Rectangle: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.